pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -838,15 +849,22 @@ func (az *Cloud) reconcileSecurityGroup(clusterName string, service *v1.Service,
 	if err != nil {
 		return nil, err
 	}
+	serviceTags, err := getServiceTags(service)
+	if err != nil {
+		return nil, err
+	}
 	var sourceAddressPrefixes []string
-	if sourceRanges == nil || serviceapi.IsAllowAll(sourceRanges) {
+	if (sourceRanges == nil || serviceapi.IsAllowAll(sourceRanges)) && len(serviceTags) == 0 {
```

Comment:
is there any possiblity that `sourceRanges != nil` and `len(serviceTags) == 0` ?

Reply:
Yep, this is expected to do actions in `else` block